src/monodroid/jni/mono-image-loader.hh

```diff
@@ -0,0 +1,148 @@
+// Dear Emacs, this is a -*- C++ -*- header
+#if !defined (__MONO_IMAGE_LOADER_HH)
+#define __MONO_IMAGE_LOADER_HH
+
+#include <cstdint>
+#include <cstddef>
+
+#include <mono/metadata/image.h>
+#if defined (NET)
+#include <mono/metadata/mono-private-unstable.h>
+#endif
+#include <mono/metadata/object.h>
+
+#include "platform-compat.hh"
+#include "xamarin-app.hh"
+#include "xxhash.hh"
+#include "search.hh"
+#include "strings.hh"
+
+#if defined (RELEASE) && defined (ANDROID) && defined (NET) && defined (ENABLE_MARSHAL_METHODS)
+#define USE_CACHE 1
+#endif
+
+namespace xamarin::android::internal {
+	enum class MonoImageLoaderContext
```

Comment:
I know large fractions of this code aren't even *built* in this PR, but why have this enum when *only* AssemblyLoadContext will be supported for marshal methods?